pkg/api/v1/types.go

```diff
@@ -1241,6 +1243,10 @@ type PodSpec struct {
 	ImagePullSecrets []LocalObjectReference `json:"imagePullSecrets,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
 }
 
+// PodSecurityContext holds pod-level security attributes and common container settings.
+type PodSecurityContext struct {
```

Comment:
@bgrant0607 @thockin That's reasonable.  I would like to keep this PR distinct and start writing changes to go in on top of this, though, if that's okay with you.